yjit/src/backend/arm64/mod.rs

```diff
@@ -351,15 +352,19 @@ impl Assembler
                     // unsigned immediate that can be encoded as a bitmask
                     // immediate. If it's not one of those, we'll need to load
                     // it first.
-                    let opnd1 = split_bitmask_immediate(asm, opnds[1]);
+                    let opnd1 = split_bitmask_immediate(&mut asm, opnds[1]);
 
                     asm.test(opnd0, opnd1);
                 },
                 _ => {
-                    asm.push_insn(op, opnds, target, text, pos_marker);
+                    asm.push_insn(insn.op, opnds, insn.target, insn.text, insn.pos_marker);
                 }
             };
-        })
+
+            iterator.map_index(&mut asm);
```

Comment:
Could this be integrated into iterator.next_* so that it happens automatically?

And if not, could you rename it to `map_insn_index` to make the meaning more explicit?

Reply:
Not in an easy way. The issue is we need this to happen after all of the new instructions have been pushed into the new assembler. So we could potentially do that when you fetch the next instruction if index > 0, but it gets a little bit weird because you're really handling the previous instruction. I'll rename it.